dropwizard-jersey/src/main/java/io/dropwizard/jersey/DropwizardResourceConfig.java

```diff
@@ -142,7 +142,7 @@ public ResourceConfig register(final Object component) {
         // If a class gets passed through as an object, cast to Class and register directly
         if (component instanceof Class<?>) {
             return super.register((Class<?>) component);
-        } else if (Providers.isProvider(clazz) || Binder.class.isAssignableFrom(clazz)) {
+        } else if (Providers.isProvider(clazz) || org.glassfish.hk2.utilities.Binder.class.isAssignableFrom(clazz)) {
```

Comment:
probably want to check both `org.glassfish.jersey.internal.inject.Binder` and `org.glassfish.hk2.utilities.Binder` if we want to add pass through for org.glassfish.hk2.utilities.Binder here?

Reply:
Can do, but looking at the first condition `Providers.isProvider(clazz)`, it is already checking for `org.glassfish.jersey.internal.inject.Binder`, see `org.glassfish.jersey.internal.inject.Providers.getExternalProviderInterfaces`. `DropwizardResourceConfigTest` also has a test that verifies that `org.glassfish.jersey.internal.inject.Binder` is registered correctly. Do you still want me to add an explicit condition for it?

Reply:
Looking through the jersey code, I agree with you that it `Providers.isProvider` should work for `org.glassfish.jersey.internal.inject.Binder` instances.  Looks like it should also work for `org.glassfish.hk2.utilities.Binder`s since that interface has the `@Contract` annotation?  As long as we can prove that it returns true for implementors of both interfaces (maybe add a test?)...

Reply:
Unfortunatly `Providers.isProvider` does not pick up `org.glassfish.hk2.utilities.Binder` .  `Providers.isProvider` is looking for `org.glassfish.jersey.spi.Contract`. However `org.glassfish.hk2.utilities.Binder`  is annotated with `org.jvnet.hk2.annotations.Contract`. 
I am adding a test `ProvidersBinderTest` that demonstrates that.

In my original commit I included  a test `DropwizardResourceConfigTest.logResourceWithHK2Binder`  that fails if I remove second condition. Also there is an existing test, `DropwizardResourceConfigTest.logEndpointBinder` that verifies that `org.glassfish.jersey.internal.inject.Binder` can be registered.